packages/react-native/Libraries/StyleSheet/processBackgroundImage.js

```diff
@@ -76,37 +91,39 @@ export default function processBackgroundImage(
         }
       }
 
-      let points: {
-        start: ParsedGradientValue['start'],
-        end: ParsedGradientValue['end'],
-      } | null = null;
-
-      if (typeof bgImage.direction === 'undefined') {
-        points = TO_BOTTOM_START_END_POINTS;
-      } else if (ANGLE_UNIT_REGEX.test(bgImage.direction)) {
-        const angle = parseAngle(bgImage.direction);
-        if (angle != null) {
-          points = calculateStartEndPointsFromAngle(angle);
+      let orientation: LinearGradientOrientation = DEFAULT_ORIENTATION;
+
+      if (
+        bgImage.direction != null &&
+        ANGLE_UNIT_REGEX.test(bgImage.direction)
+      ) {
+        const parsedAngle = getAngleInDegrees(bgImage.direction);
+        if (parsedAngle != null) {
+          orientation = {
+            type: 'angle',
+            value: parsedAngle,
+          };
         }
-      } else if (DIRECTION_REGEX.test(bgImage.direction)) {
-        const processedPoints = calculateStartEndPointsFromDirection(
-          bgImage.direction,
-        );
-        if (processedPoints != null) {
-          points = processedPoints;
+      } else if (
+        bgImage.direction != null &&
+        DIRECTION_REGEX.test(bgImage.direction)
+      ) {
+        const parsedDirection = getDirectionString(bgImage.direction);
+        if (parsedDirection != null) {
+          orientation = {
```

Comment:
I thought about this too but it seems keywords like `to top left` (which means the angle should align with the view's corner) need information about the view which we can't do during parsing.

We could however preemptively parse `to top`, `to left` etc since those map to constant angle values. However I feel like we should avoid parsing in such a fragmented way

Reply:
i also had same thoughts as @jorge-cab about parsing in fragmented way. i was also thinking in the longer term that dimension-dependent styles like these could be part of layout nodes right before setting native view dimensions, lot of native styling logic could be moved there and code can be reused among platforms. e.g transform-origin, translate with %, gradients, also support for calc function etc. Is there an ongoing effort in that direction? i can also take a look, i know @NickGerleman is working on native CSS parser, but i haven't dug very deep into it yet.

Reply:
Updated [here](https://github.com/facebook/react-native/pull/47003/commits/d308b7d501e7343a58f041e91d3f925e48a6efdb), loving it. It made it more concise and reusable.